var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -55,13 +55,16 @@ class Hdf5(AutotoolsPackage):
     variant('shared', default=True,
             description='Builds a shared version of the library')
 
-    variant('cxx', default=True, description='Enable C++ support')
-    variant('fortran', default=True, description='Enable Fortran support')
+    variant('unsupported', default=False,
```

Comment:
I am not in favor of this. This variant does NOT change package behavior anyhow, as discussed in the issue u linked

Reply:
To clarify: If I understand correctly, say with default settings, hdf5~unsupported and hdf5+unsupported are exactly the same builds. No functional difference. Thus it shall not be a variant and was removed.

Reply:
I don't insist on this but why are there 'conflicts' statements in the current script and how can I tell spack to ignore them? For example, if I want to enable the cxx interface and have the thread safety. The developers of hdf5 allow for that. Of course, I have to take the responsibility explicitly (--enable-unsupported) in that case. The 'unsupported' variant is for the same reason: to allow package developers (and users) to specify that they need/want an unsupported combination of features and they know what they are doing. Otherwise, they get a notification from the 'conflicts' statement.

Reply:
Then just remove the conflicts statements. Probably they should have been removed earlier together with unsupported. 

One way or another, variants are meant to change the package, which is not the case here. Similarly “run-tests” is not a variant, instead we have —run-tests option of install command

Reply:
@skosukhin For what is worth, my interpretation at the time I did this was:
- `+cxx+mpi` is claimed to be *unsupported* -> there's no promise on its behavior
- `+cxx+threadsafe` is claimed to be *not compatible* ->  it's known to be broken

Reply:
> +cxx+mpi is claimed to be unsupported -> there's no promise on its behavior

that we can probably allow, but maybe print a warning to the user during installation.

> +cxx+threadsafe is claimed to be not compatible -> it's known to be broken

this this shall stay as a conflict.

Reply:
> +cxx+threadsafe is claimed to be not compatible -> it's known to be broken

@alalazo , does this mean that the 'threadsafe' feature doesn't work even via low-level interface (when +cxx) or only via cxx interface?

Reply:
@skosukhin I would expect only through the `cxx` interface (and I hope the low-level is compiled the same regardless of `cxx`). Don't take my word for it, though - never peeked the code inside `hdf5`, and the docs are not 100% clear.